examples_tests/20.Megatexture/main.cpp

```diff
@@ -931,51 +585,71 @@ int main()
+    auto gpuds2 = driver->createGPUDescriptorSet(std::move(gpu_ds2layout));
+    core::smart_refctd_ptr<video::IGPUBuffer> ds2_ssbo;
     {
         constexpr uint32_t ENTRY_SZ = sizeof(uint32_t);
         constexpr uint32_t ENTRY_ALIGNMENT = sizeof(uint32_t);//change to 16 if std140
         constexpr uint32_t PRECOMPUTED_STUFF_COUNT = 3u;//pgtab_sz_log2, phys_pg_tex_sz_rcp, vtex_sz_rcp
-        constexpr uint32_t PRECOMPUTED_STUFF_SZ = ETP_COUNT*ENTRY_ALIGNMENT;
-        constexpr uint32_t UBO_SZ = PRECOMPUTED_STUFF_COUNT*PRECOMPUTED_STUFF_SZ;
-        uint8_t ubodata[UBO_SZ]{};
-        uint32_t* ptr = reinterpret_cast<uint32_t*>(ubodata);
+        constexpr uint32_t PRECOMPUTED_STUFF_SZ = VT_COUNT*ENTRY_ALIGNMENT;
+        constexpr uint32_t PRECOMPUTED_SSBO_SZ = PRECOMPUTED_STUFF_COUNT*PRECOMPUTED_STUFF_SZ;
+        uint8_t precomputedStuff[PRECOMPUTED_SSBO_SZ]{};
+        uint32_t* ptr = reinterpret_cast<uint32_t*>(precomputedStuff);
         uint32_t offset = 0u;
         //pgtab_sz_log2
-        for (uint32_t i = 0u; i < ETP_COUNT; ++i)
-            ptr[i*ENTRY_ALIGNMENT/ENTRY_SZ] = core::findMSB(texPackers[i]->getPageTable()->getCreationParameters().extent.width);
+        for (uint32_t i = 0u; i < VT_COUNT; ++i)
+            ptr[i*ENTRY_ALIGNMENT/ENTRY_SZ] = core::findMSB(gpuvt->getPageTable()->getCreationParameters().extent.width);
         ptr += PRECOMPUTED_STUFF_SZ/ENTRY_SZ;
         //phys_pg_tex_sz_rcp
-        for (uint32_t i = 0u; i < ETP_COUNT; ++i)
+        for (uint32_t i = 0u; i < VT_COUNT; ++i)
         {
-            const double f = 1.0 / static_cast<double>(texPackers[i]->getPhysicalAddressTexture()->getCreationParameters().extent.width);
+            const auto& storageImg = gpuvt->getFloatViews().begin()[i]->getCreationParameters().image;
+            const double f = 1.0 / static_cast<double>(storageImg->getCreationParameters().extent.width);
             reinterpret_cast<float*>(ptr)[i*ENTRY_ALIGNMENT/ENTRY_SZ] = f;
         }
         ptr += PRECOMPUTED_STUFF_SZ/ENTRY_SZ;
         //vtex_sz_rcp
-        for (uint32_t i = 0u; i < ETP_COUNT; ++i)
+        for (uint32_t i = 0u; i < VT_COUNT; ++i)
         {
             double f = 1.0;
-            f /= static_cast<double>(texPackers[i]->getPageTable()->getCreationParameters().extent.width);
-            f /= static_cast<double>(texPackers[i]->getPageSize());
+            f /= static_cast<double>(gpuvt->getPageTable()->getCreationParameters().extent.width);
+            f /= static_cast<double>(gpuvt->getPageExtent());
             reinterpret_cast<float*>(ptr)[i*ENTRY_ALIGNMENT/ENTRY_SZ] = f;
         }
-        gpuPrecomputedStuffSsbo = driver->createFilledDeviceLocalGPUBufferOnDedMem(UBO_SZ,ubodata);
+
+        const uint32_t lutSz = gpuvt->getLayerToViewIndexLUTBytesize();
+        uint32_t lut[PGTAB_LAYERS]{};
+        assert(sizeof(lut)>=lutSz);
+        gpuvt->writeLayerToViewIndexLUTContents(lut);
```

Comment:
something tells me that returning a `const uint32_t*`  + size would be a lot cleaner and less lines of code than this